builtin/providers/openstack/resource_openstack_compute_instance_v2.go

```diff
@@ -659,3 +671,36 @@ func getFloatingIPs(networkingClient *gophercloud.ServiceClient) ([]floatingips.
 	}
 	return ips, nil
 }
+
+func getImageID(client *gophercloud.ServiceClient, d *schema.ResourceData) (string, error) {
+	imageID := d.Get("image_id").(string)
+	imageName := d.Get("image_name").(string)
+	if imageID == "" {
+		pager := images.ListDetail(client, nil)
```

Comment:
Also, image name is not unique in Glance. What if there are multiple images with the desired name ? With your implementation it will silently take one of them. In my opinion, in this case it would be better to fail we an explicit message.